drivers/led/ncp5623c.c

```diff
@@ -0,0 +1,208 @@
+		return -EINVAL;
+	}
+
+	uint8_t buf[6] = {0x70, NCP5623_LED_PWM0, 0x70, NCP5623_LED_PWM1, 0x70, NCP5623_LED_PWM2};
+
+	buf[1] = buf[1] | color[0] / 8;
+	buf[3] = buf[3] | color[1] / 8;
+	buf[5] = buf[5] | color[2] / 8;
+
+	return i2c_burst_write_dt(&config->bus, NCP5623_LED_CURRENT | NCP5623C_MAX_BRIGHTNESS, buf,
+				  sizeof(buf));
+}
+
+static int ncp5623c_led_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
+{
+	const struct ncp5623c_config *config = dev->config;
+	struct ncp5623c_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+	int ret;
+
+	if (led > (sizeof(led_channels) - 1)) {
+		return -EINVAL;
+	}
+
+	/* Recale 0..100 to 0..31 */
+
+	value = value / 3;
+
+	if (value > dev_data->max_brightness) {
+		value = dev_data->max_brightness;
+	}
+
+	ret = i2c_reg_write_byte_dt(&config->bus, led_channels[led] | value, 0x70);
+
+	if (ret < 0) {
+		LOG_ERR("LED write failed");
+		return ret;
+	}
+
+	return 0;
+}
+
+static inline int ncp5623c_led_on(const struct device *dev, uint32_t led)
+{
+	struct ncp5623c_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+
+	return ncp5623c_led_set_brightness(dev, led, dev_data->max_brightness);
+}
+
+static inline int ncp5623c_led_off(const struct device *dev, uint32_t led)
+{
+	struct ncp5623c_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+
+	return ncp5623c_led_set_brightness(dev, led, dev_data->min_brightness);
+}
+
+static int ncp5623c_led_init(const struct device *dev)
+{
+	const struct ncp5623c_config *config = dev->config;
+	struct ncp5623c_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
+
+	/* Hardware specific limits */
+	dev_data->min_brightness = NCP5623C_MIN_BRIGHTNESS;
+	dev_data->max_brightness = NCP5623C_MAX_BRIGHTNESS;
+
+	uint8_t msg[6] = {0x70, NCP5623_LED_PWM0, 0x70, NCP5623_LED_PWM1, 0x70, NCP5623_LED_PWM2};
+
+	if (i2c_burst_write_dt(&config->bus, NCP5623_LED_CURRENT | NCP5623C_MAX_BRIGHTNESS, msg,
+			       6)) {
+		LOG_ERR("LED write failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static const struct led_driver_api ncp5623c_led_api = {
+	.set_brightness = ncp5623c_led_set_brightness,
+	.on = ncp5623c_led_on,
+	.off = ncp5623c_led_off,
+	.get_info = ncp5623c_get_info,
+	.set_color = ncp5623c_set_color,
+};
+
+#define COLOR_MAPPING(led_node_id)                                                                 \
+	static const uint8_t color_mapping_##led_node_id[] = DT_PROP(led_node_id, color_mapping);
+
+#define LED_INFO(led_node_id)                                                                      \
+	{                                                                                          \
+		.label = DT_PROP(led_node_id, label),                                              \
+		.index = DT_PROP(led_node_id, index),                                              \
```

Comment:
Well... Since there is only one LED on the controller, then maybe you don't need the `index` property to find the index. 0 is a good guess :)